src/System.Private.ServiceModel/tests/Scenarios/Client/ExpectedExceptions/ExpectedExceptionTests.cs

```diff
@@ -286,10 +287,37 @@ public static void UnknownUrl_Throws_EndpointNotFoundException()
 
         EndpointNotFoundException exception = Assert.Throws<EndpointNotFoundException>(() =>
         {
-            using (ChannelFactory<IWcfService> factory = new ChannelFactory<IWcfService>(binding, new EndpointAddress(notFoundUrl)))
+            try
             {
-                IWcfService serviceProxy = factory.CreateChannel();
-                string response = serviceProxy.Echo("Hello");
+                using (
+                    ChannelFactory<IWcfService> factory = new ChannelFactory<IWcfService>(binding,
+                        new EndpointAddress(notFoundUrl)))
+                {
+                    IWcfService serviceProxy = factory.CreateChannel();
+                    string response = serviceProxy.Echo("Hello");
+                }
+            }
+            catch (EndpointNotFoundException)
+            {
+                throw;
+            }
```

Comment:
I believe this catch/throw is needed to avoid the lower catch but still preserve the Assert.Throws<T> usage